sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/SchemaAndRecord.java

```diff
@@ -33,6 +35,11 @@ public SchemaAndRecord(GenericRecord record, TableSchema tableSchema) {
     this.tableSchema = tableSchema;
   }
 
+  public SchemaAndRecord(Row record, TableSchema tableSchema) {
```

Comment:
nit: let's not do this conversion in the `SchemaAndRecord` constructor. Instead we could change `BigQueryStorageStreamSource.readSingleRecord` to always return a `GenericRecord`, and the Arrow implementation can do the conversion there.